rcl/src/rcl/logging_rosout.c

```diff
@@ -243,20 +246,31 @@ rcl_ret_t rcl_logging_rosout_init_publisher_for_node(rcl_node_t * node)
   // Verify input and make sure it's not already initialized
   RCL_CHECK_ARGUMENT_FOR_NULL(node, RCL_RET_NODE_INVALID);
   logger_name = rcl_node_get_logger_name(node);
+  const char * node_name = rcl_node_get_name(node);
   if (NULL == logger_name) {
     RCL_SET_ERROR_MSG("Logger name was null.");
     return RCL_RET_ERROR;
   }
   if (rcutils_hash_map_key_exists(&__logger_map, &logger_name)) {
     // @TODO(nburek) Update behavior to either enforce unique names or work with non-unique
     // names based on the outcome here: https://github.com/ros2/design/issues/187
+    const char * first_part = "Publisher already registered for: ";
+    const char * second_part = ". If this is due to multiple nodes with the same name "
+      "then all logs for: ";
+    const char * third_part = " will go out over the existing publisher. As soon as any "
+      "Node with that name is destructed it will unregister the publisher, preventing a "
+      "any further logs for that name from being published on the rosout topic.";
+
+    const size_t MsgSize = strlen(logger_name) + strlen(node_name) +
+      strlen(first_part) + strlen(second_part) + strlen(third_part) + 1;
+    char message[MsgSize];
```

Comment:
This isn't very consistent with our style.  This should be something like:

```suggestion
    const size_t msg_size = strlen(logger_name) + strlen(node_name) +
      strlen(first_part) + strlen(second_part) + strlen(third_part) + 1;
    char message[msg_size];
```

(also, today I learned that `strlen` on a constant string is actually evaluated at compile time, not runtime.  Neat!)

Reply:
I figure it should work here since everything is known at compile time and the code doesn't throw any warning/error, but `cpplint` complains about using a variable array with that style for some reason. Test looks like this:
> logging_rosout.c:270:  Do not use variable-length arrays. Use an appropriately named ('k' followed by CamelCase) compile-time constant for the size

Reply:
Oh, but the `node_name` and `logger_name` is not known at compile-time, which is why it is complaining.

Reply:
In that case, should I just make `msg_size` const, by removing `strlen(node_name)` and `strlen(logger_name)`, then replacing that will a constant value like 100. It should be enough for both the logger and node name for the most part, while not being too large. This takes away a variable sized array, because it would be all constant.

Reply:
No, I think we should just allocate on the heap here.